src/palace/manager/celery/tasks/identifiers.py

```diff
@@ -0,0 +1,159 @@
+
+    Any identifiers present in the existing set but not in the active set will be marked as unavailable in
+    the collection. This is done by sending a circulation_apply task that sets both licenses_available and
+    licenses_owned to 0 for each identifier in the collection.
+
+    This function is designed to be used as the body of a chord created by `create_mark_unavailable_chord`.
+    """
+    redis_client = task.services.redis().client()
+
+    existing_identifiers, active_identifiers = identifiers
+    existing_set = IdentifierSet(redis_client, **existing_identifiers)
+    active_set = IdentifierSet(redis_client, **active_identifiers)
+
+    try:
+        if not existing_set.exists():
+            task.log.warning(
+                "Existing identifiers set does not exist in Redis. No identifiers to mark as unavailable."
+            )
+            return
+
+        if not active_set.exists():
+            task.log.error(
+                "Active identifiers set does not exist in Redis. Refusing to mark all identifiers as unavailable."
+            )
+            raise PalaceValueError("Active identifiers set does not exist in Redis.")
+
+        with task.session() as session:
+            collection = load_from_id(session, Collection, collection_id)
+            data_source_name = validate_not_none(
+                collection.data_source,
+                message="Collection has no data source! (data_source is None).",
+            ).name
+            collection_name = collection.name
+
+        create_circulation_data = partial(
+            CirculationData,
+            data_source_name=data_source_name,
+            licenses_owned=0,
+            licenses_available=0,
+        )
+        identifiers_to_mark = existing_set - active_set
+        for identifier in identifiers_to_mark:
+            task.log.info(
+                f"Marking identifier {identifier} as unavailable in collection {collection_name} ({collection_id})"
+            )
+            circulation_apply.delay(
+                circulation=create_circulation_data(primary_identifier_data=identifier),
+                collection_id=collection_id,
+            )
+
+        task.log.info(
+            f"Sent tasks to mark {len(identifiers_to_mark)} identifiers as unavailable"
+        )
+    finally:
+        existing_set.delete()
+        active_set.delete()
+
+
+def create_mark_unavailable_chord(
+    collection_id: int, active_identifiers_sig: Signature
+) -> Signature:
+    """
+    Creates a Celery chord that identifies and marks as unavailable any identifiers that were not
+    found in the distributor's feed for a given collection.
+
+    This chord performs the following sequence:
+    1. Calls the existing_available_identifiers task to retrieve all identifiers that are currently
+       available in the collection.
+    2. In parallel, executes the provided active_identifiers_sig task to retrieve identifiers that
+       are present in the distributor's feed.
```

Comment:
Minor suggestion: Considering "numbering" these as "1.a." and "1.b", since they are meant to be running in parallel. Something like:

```
    1. Performs the following in parallel:
      a. Calls the existing_available_identifiers task to retrieve all identifiers that are currently
       available in the collection.
      b. Executes the provided active_identifiers_sig task to retrieve identifiers that
       are present in the distributor's feed.
    2. Finally, ...
```

Reply:
Updated